sphinx_qt_documentation/__init__.py

```diff
@@ -21,14 +21,26 @@
 from typing import List, Optional, Dict, Any
 from sphinx.locale import get_translation
 from sphinx.ext.intersphinx import InventoryAdapter
-from qtpy.QtCore import Signal
 
 _ = get_translation("sphinx")
 
-try:
-    from qtpy import QT_VERSION
-except ImportError:
-    QT_VERSION = None
+
+def _get_signal_and_version():
+    version_mapping = {'qtpy': lambda: 'QT_VERSION', 'Qt': lambda: '__qt_version__',
+                       'Pyside2': lambda: importlib.import_module('PySide2.QtCore').__version__,
+                       'PyQt5': lambda: importlib.import_module('PyQt5.QtCore').QT_VERSION_STR}
+    for module_name in version_mapping.keys():
+        try:
+            core = importlib.import_module(f'{module_name}.QtCore')
+            signal = getattr(core, 'Signal')
```

Comment:
in PyQt5 it is not `Signal`, but `pyqtSignal`

Reply:
D'oh. Good thing these abstraction shims exist :wink: